app/src/main/java/org/connectbot/HostListActivity.java

```diff
@@ -273,6 +276,17 @@ public boolean onMenuItemClick(MenuItem item) {
 		colors.setIcon(android.R.drawable.ic_menu_slideshow);
 		colors.setIntent(new Intent(HostListActivity.this, ColorsActivity.class));
 
+		disconnectall = menu.add(R.string.list_menu_disconnect);
+		disconnectall.setIcon(android.R.drawable.ic_menu_delete);
+		final HostListActivity self = this;
+		disconnectall.setOnMenuItemClickListener(new OnMenuItemClickListener() {
+			@Override
+			public boolean onMenuItemClick(MenuItem menuItem) {
+				self.disconnectAll();
```

Comment:
You can just call `disconnectAll()`. No need for the self stuff. See the calls to `updateList` in the above handlers.